sdk/azidentity/managed_identity_client.go

```diff
@@ -179,7 +180,12 @@ func (c *managedIdentityClient) authenticate(ctx context.Context, id ManagedIDKi
 		if id != nil {
 			return azcore.AccessToken{}, newAuthenticationFailedError(credNameManagedIdentity, "the requested identity isn't assigned to this resource", resp, nil)
 		}
-		return azcore.AccessToken{}, newCredentialUnavailableError(credNameManagedIdentity, "no default identity is assigned to this resource")
+		msg := "failed to authenticate a system assigned identity"
+		if body, err := io.ReadAll(resp.Body); err == nil && len(body) > 0 {
+			msg += fmt.Sprintf(". The endpoint responded with %s", body)
+		}
+		resp.Body.Close()
```

Comment:
I believe using this:

https://github.com/Azure/azure-sdk-for-go/blob/514985a4c9d311c90b86c74c6ff779abcec1ffcd/sdk/azcore/runtime/response.go#L24

Will make sure the response body is still available, which is useful if they're using [`WithCaptureResponse`](https://github.com/Azure/azure-sdk-for-go/blob/514985a4c9d311c90b86c74c6ff779abcec1ffcd/sdk/azcore/runtime/policy_include_response.go#L32).

Reply:
Thanks, that's definitely better